packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/DynamicZone/Field.tsx

```diff
@@ -0,0 +1,279 @@
+import * as React from 'react';
+
+import { Box, Flex, VisuallyHidden } from '@strapi/design-system';
+import { useNotification } from '@strapi/helper-plugin';
+import pipe from 'lodash/fp/pipe';
+import { useIntl } from 'react-intl';
+
+import { createContext } from '../../../../../../components/Context';
+import { useField, useForm } from '../../../../../components/Form';
+import { useDoc } from '../../../../../hooks/useDocument';
+import { type EditFieldLayout } from '../../../../../hooks/useDocumentLayout';
+import { getTranslation } from '../../../../../utils/translations';
+import { transformDocument } from '../../../utils/data';
+import { createDefaultForm } from '../../../utils/forms';
+
+import { AddComponentButton } from './AddComponentButton';
+import { ComponentPicker } from './ComponentPicker';
+import { DynamicComponent, DynamicComponentProps } from './DynamicComponent';
+import { DynamicZoneLabel } from './DynamicZoneLabel';
+
+import type { Attribute } from '@strapi/types';
+
+interface DynamicZoneContextValue {
+  isInDynamicZone: boolean;
+}
+
+const [DynamicZoneProvider, useDynamicZone] = createContext<DynamicZoneContextValue>(
+  'DynamicZone',
+  {
+    isInDynamicZone: false,
+  }
+);
+
+interface DynamicZoneProps
+  extends Omit<Extract<EditFieldLayout, { type: 'dynamiczone' }>, 'size'> {}
+
+const DynamicZone = ({
+  attribute,
+  disabled,
+  hint,
+  label,
+  name,
+  required = false,
+}: DynamicZoneProps) => {
+  // We cannot use the default props here
+  const { max = Infinity, min = -Infinity } = attribute ?? {};
+
+  const [addComponentIsOpen, setAddComponentIsOpen] = React.useState(false);
+  const [liveText, setLiveText] = React.useState('');
+  const { components } = useDoc();
+  const { addFieldRow, removeFieldRow, moveFieldRow } = useForm(
+    'DynamicZone',
+    ({ addFieldRow, removeFieldRow, moveFieldRow }) => ({
+      addFieldRow,
+      removeFieldRow,
+      moveFieldRow,
+    })
```

Comment:
I don't understand this, it looks like it could be `(args) => args`

Reply:
it's not, because its a selector for the `useForm` hook so the arguments are a much bigger object